From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 6 Aug 2010 14:23:10 -0400
Subject: [net] qlge: fix a eeh handler to not add a pending timer
Message-id: <20100806141957.2774.69491.sendpatchset@localhost.localdomain>
Patchwork-id: 27451
O-Subject: [RHEL 5.6 PATCH 9/10] qlge: fix a eeh handler to not add a pending
	timer.
Bugzilla: 567402

Bugzilla
========

567402

Upstream Status
===============

net-2.6 commit id 7ae80abdba0644e12ac17da567a2db1efc1bf8a8

Description
===========

qlge: Update driver version to 1.00.00.24
>From 497f09af396bf4d3373801528ce7f8a8493866c6 Mon Sep 17 00:00:00 2001
From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Wed, 4 Aug 2010 17:49:20 -0700
Subject: [PATCH 1/2] 	 qlge: fix a eeh handler to not add a pending timer.

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>

diff --git a/drivers/net/qlge/qlge_main.c b/drivers/net/qlge/qlge_main.c
index e4187c4..5811aef 100644
--- a/drivers/net/qlge/qlge_main.c
+++ b/drivers/net/qlge/qlge_main.c
@@ -4626,6 +4626,9 @@ static void ql_eeh_close(struct net_device *ndev)
 		netif_stop_queue(ndev);
 	}
 
+	/* Disabling the timer */
+	del_timer_sync(&qdev->eeh_timer);
+
 	if (test_bit(QL_ADAPTER_UP, &qdev->flags))
 		cancel_delayed_work_sync(&qdev->asic_reset_work);
 	cancel_delayed_work_sync(&qdev->mpi_reset_work);
